go-fuzz/main.go

```diff
@@ -88,5 +99,31 @@ func main() {
 		go slaveMain()
 	}
 
+	if *flagMasterStats != "" {
+		evMux := http.NewServeMux()
+		evMux.HandleFunc("/eventsource", func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Type", "text/event-stream")
+			w.WriteHeader(http.StatusOK)
+			<-writerSet.Add(w)
+		})
+		evMux.HandleFunc("/stats", func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Type", "text/html")
+			w.WriteHeader(http.StatusOK)
+			io.WriteString(w, statsPage)
+		})
+
+		go http.ListenAndServe(*flagMasterStats, evMux)
+	}
+
 	select {}
 }
+
+func broadcastStats(stats masterStats) {
+	b, err := json.Marshal(stats)
+	if err != nil {
+		panic(err)
```

Comment:
It's a fairly simple struct type and marshaling it should *never* cause an actual error unless somebody writes in a related bug, so a panic seemed to make sense. The only sensical alternative I could see is logging the error and skip writing to the output.